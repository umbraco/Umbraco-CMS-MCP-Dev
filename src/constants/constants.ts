export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
