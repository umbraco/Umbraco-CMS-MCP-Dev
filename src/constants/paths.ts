import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');