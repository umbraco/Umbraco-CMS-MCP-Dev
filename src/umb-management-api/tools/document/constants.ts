export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}