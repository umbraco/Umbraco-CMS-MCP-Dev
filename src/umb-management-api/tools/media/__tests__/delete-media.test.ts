import DeleteMediaTool from "../delete/delete-media.js";
import { MediaBuilder } from "./helpers/media-builder.js";
import { MediaTestHelper } from "./helpers/media-test-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";
import { TemporaryFileBuilder } from "../../temporary-file/__tests__/helpers/temporary-file-builder.js";

const TEST_MEDIA_NAME = "_Test Media Delete";

describe("delete-media", () => {
  let originalConsoleError: typeof console.error;
  let tempFileBuilder: TemporaryFileBuilder;

  beforeEach(async () => {
    originalConsoleError = console.error;
    console.error = jest.fn();

    tempFileBuilder = await new TemporaryFileBuilder()
      .withExampleFile()
      .create();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    // Clean up any remaining test media
    await MediaTestHelper.cleanup(TEST_MEDIA_NAME);
  });

  it("should delete a media item", async () => {
    // Create a media item to delete
    const builder = await new MediaBuilder()
      .withName(TEST_MEDIA_NAME)
      .withImageMediaType()
      .withImageValue(tempFileBuilder.getId())
      .create();

    // Delete the media
    const result = await DeleteMediaTool().handler(
      {
        id: builder.getId(),
      },
      { signal: new AbortController().signal }
    );

    // Verify the handler response using snapshot
    expect(result).toMatchSnapshot();

    // Verify the media no longer exists
    const found = await MediaTestHelper.findMedia(TEST_MEDIA_NAME);
    expect(found).toBeUndefined();
  });

  it("should handle non-existent media", async () => {
    const result = await DeleteMediaTool().handler(
      {
        id: BLANK_UUID,
      },
      { signal: new AbortController().signal }
    );

    // Verify the error response using snapshot
    expect(result).toMatchSnapshot();
  });
});
