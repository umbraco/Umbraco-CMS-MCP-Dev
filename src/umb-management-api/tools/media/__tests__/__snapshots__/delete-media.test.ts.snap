// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete-media should delete a media item 1`] = `
{
  "content": [
    {
      "text": """",
      "type": "text",
    },
  ],
}
`;

exports[`delete-media should handle non-existent media 1`] = `
{
  "content": [
    {
      "text": "Error using delete-media:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "The content could not be found",
    "status": 404,
    "operationStatus": "NotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;
